src/data/hooks/useApy.ts

```diff
@@ -1,50 +1,64 @@
 import { useQuery } from "@tanstack/react-query"
-import { getApy as getApy_AAVE_V2_STABLE_CELLAR } from "data/actions/CELLAR_V0815/getApy"
 import { CellarNameKey, ConfigProps } from "data/types"
-import { CellarStakingV0815, CellarV0815 } from "src/abi/types"
+import { useGet10DaysShareValueQuery } from "generated/subgraph"
+import { getPrevious10Days } from "utils/calculateTime"
 import { useCreateContracts } from "./useCreateContracts"
-import { useSommelierPrice } from "./useSommelierPrice"
+
+const previous10Days = getPrevious10Days()
 
 export const useApy = (config: ConfigProps) => {
   const { cellarContract, stakerContract } =
     useCreateContracts(config)
 
-  const sommPrice = useSommelierPrice()
+  const [{ fetching, data, error }, reexecute10Days] =
+    useGet10DaysShareValueQuery({
+      variables: {
+        epoch: previous10Days,
+        cellarAddress: config.id,
+      },
+    })
+
+  const latestData = data?.cellar?.dayDatas.at(-1)
```

Comment:
Instead of the latest data, we want to detect the date where the `shareValue` changed. Since it only happens when accrue is called and that happens once a week right now. The most recent day and the previous day might be the same.

Something like:
```js
const found = dayDatas.findIndex((data, idx, arr) => {
  if (idx === 0)  return false;
  
  const prev = arr[idx - 1].shareValue;
  return prev === data.shareValue;
});

let apy = 0;
if (found > 0) {
  // calc apy
}
```